common/ferrostar/src/routing_adapters/osrm/models.rs

```diff
@@ -87,14 +87,40 @@ pub struct Annotation {
     /// NOTE: This annotation is not in the official spec, but is a common extension used by Mapbox
     /// and Valhalla.
     #[serde(default, rename = "maxspeed")]
-    #[allow(dead_code)]
-    max_speed: Vec<MaxSpeed>,
+    pub max_speed: Vec<MaxSpeed>,
+}
+
+/// A flattened OSRM flavored annotation object.
+/// This represents the value for each annotation type and is typically surfaced for each
+/// geometry segment.
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct AnnotationValue {
+    pub distance: f64,
+    pub duration: f64,
+    /// The speed limit for this geometry segment in meters per second.
+    /// If the speed limit is unknown, this will be `None`.
+    pub max_speed_mps: Option<f64>,
+    /// The expected travel speed.
+    pub speed_mps: f64,
```

Comment:
Minor edits. I think we can make our APIs clearer than the original ;) Also we can probably use `f32` in more places. It's often more efficient to compute on, and definitely more efficient to store.

I'm also tempted to to something like `type MetersPerSecond = f32` or even a newtype (though those might introduce annoyance serializing over the FFI).

```suggestion
    pub distance: f32,
    pub duration: f32,
    /// The speed limit for this geometry segment in meters per second.
    /// If the speed limit is unknown, this will be `None`.
    pub speed_limit: Option<f32>,
    /// The estimated travel speed, in meters/second.
    pub estimated_speed: f32,
```